clang/lib/CodeGen/Targets/NVPTX.cpp

```diff
@@ -292,7 +292,7 @@ void NVPTXTargetCodeGenInfo::setTargetAttributes(
   }
 
   // Attach kernel metadata directly if compiling for NVPTX.
-  if (FD->hasAttr<NVPTXKernelAttr>()) {
+  if (FD->hasAttr<DeviceKernelAttr>()) {
     F->setCallingConv(llvm::CallingConv::PTX_Kernel);
```

Comment:
This is now set in the 2 language paths,, there's additional code that can be folded now

Reply:
Should be done in latest commit!